### PR TITLE
BUG: Fixed output variable overriding in numpy.info()

### LINKED ISSUE #20423: numpy.info interacts poorly with sys.stdout overrides
`numpy.info` uses `sys.stdout` as a default parameter. If `sys.stdout` gets replaced with a new object, `numpy.info` ends up still trying to write to the old `sys.stdout` for any call that doesn't pass an explicit `output` parameter. For example, if you call `numpy.info` in Google Colab, you don't get any info output, because everything gets written to the wrong output stream. (I don't know if the behavior is any different in a regular Jupyter notebook - I don't have Jupyter installed.)

It would be better if `numpy.info` instead used `None` as the default parameter, and did

```
if output is None:
    output = sys.stdout
```

inside the function body, to pick up `sys.stdout` at the time of the call instead of at definition time.

### PR DESCRIPTION
**Pull request by abatomunkuev**
Closes #20423. 

## What does this implement/fix?
`numpy.info()` currently overrides the `output` variable, so that we are not able to see the output in Google Colab.
image

This is due to the use of `sys.stdout` as a default value for `output` parameter. So, I have implemented the changes that were proposed in issue #20423. Specifically, I have changed the default value to `None`, and made a check if the value of parameter `output` is `None`, then set the `output` value to `sys.stdout`. 

## What changes have been made?
- Changed default value of a parameter `output` to `None`.
- Added additional check that sets the value of `output` variable to `sys.stdout`, if it is `None`. 
- Checked if the changes are working in Google Colab
Screen Shot 2021-11-23 at 6 34 25 PM

### CODE REVIEW

File: numpy/lib/utils.py [hunk 1]
```diff
@@ -496,7 +499,7 @@ def info(object=None, maxwidth=76, output=sys.stdout, toplevel='numpy'):
         Printing width.
     output : file like object, optional
         File like object that the output is written to, default is
-        ``stdout``.  The object has to be opened in 'w' or 'a' mode.
+        ``None``.  The object has to be opened in 'w' or 'a' mode.
```
Comment (mattip) [inline numpy/lib/utils.py:502]:
```suggestion
        ``None``, in which case ``sys.stdout`` will be used.
         The object has to be opened in 'w' or 'a' mode.
```

Comment (mattip) [reply numpy/lib/utils.py:502]: One nit in the docstring

Comment (abatomunkuev) [reply numpy/lib/utils.py:502]: Thank you for review. I am going to fix it.

**Comment by mattip** [comment]
Thanks @abatomunkuev

**Pull request merged**